### ISSUE TITLE
Error message when using invalid YAML config file

### ISSUE THREAD

**Issue by ekamara** [charleskorn/batect#13]
When I incorrectly indent one line in the batect.yml, I get the following error:

```
batect.yml (line 20, column 3): while parsing a block mapping
 in 'reader', line 1, column 1:
    project_name: international-tran ... 
    ^
expected <block end>, but found BlockMappingStart
 in 'reader', line 20, column 3:
      database:
      ^

Caused by: com.fasterxml.jackson.dataformat.yaml.snakeyaml.error.MarkedYAMLException: while parsing a block mapping
 in 'reader', line 1, column 1:
    project_name: international-tran ... 
    ^
expected <block end>, but found BlockMappingStart
 in 'reader', line 20, column 3:
      database:
      ^

 at [Source: (sun.nio.ch.ChannelInputStream); line: 20, column: 3]
```

**Comment by charleskorn** [comment]
Yep, that error message could be much better. Thanks for the feedback @ekamara!

**Comment by charleskorn** [comment]
This should be fixed with 99c45a7.

Thanks again for the feedback @ekamara.
